server/controllers/medical/patients/checkin.js

```diff
@@ -102,3 +104,81 @@ function create(req, res, next) {
     .catch(next)
     .done();
 }
+
+/*
+Search for information about the latest bill
+*/
+function latest(req, res, next) {
+  const uid = db.bid(req.params.uuid);
+  var invoiceLatest;
+
+  let sql =
+    `SELECT invoice.uuid, invoice.debtor_uuid, invoice.date, CONCAT(user.first, user.last) as user,
+     invoice.cost
+    FROM invoice 
+    JOIN user ON user.id = invoice.user_id
+    WHERE debtor_uuid = ?
+    ORDER BY date DESC
+    LIMIT 1`;
+
+  db.exec(sql, [uid])
+    .then(function (result) {
+      invoiceLatest = result[0];
+      var uuid = invoiceLatest.uuid;
+
+    sql =
+      `SELECT BUID(i.uuid) as uid, CONCAT(project.abbr, invoice.reference) as reference,
+        credit, debit, (debit - credit) as balance, BUID(entity_uuid) as entity_uuid
+      FROM (
+        SELECT uuid, SUM(debit) as debit, SUM(credit) as credit, entity_uuid
+        FROM (
+          SELECT record_uuid as uuid, debit, credit, entity_uuid
+          FROM combined_ledger
+          WHERE record_uuid IN (?) AND entity_uuid = ?
+        UNION ALL
+          SELECT reference_uuid as uuid, debit, credit, entity_uuid
+          FROM  combined_ledger
+          WHERE reference_uuid IN (?) AND entity_uuid = ?
+        ) AS ledger
+        GROUP BY entity_uuid
+      ) AS i JOIN invoice ON i.uuid = invoice.uuid
+      JOIN project ON invoice.project_id = project.id `;
+
+    var sql2 =
+      `SELECT COUNT(BUID(i.uuid)) as numberPayment
```

Comment:
Instead of using `BUID()` in the SQL statement, use the javascript function `db.bid()`.  The `BUID()` function in SQL converts from BINARY to STRING, while the `db.bid()` converts from STRING to BINARY.